tests/DIConfigServiceProviderTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace tests\TomPHP\ConfigServiceProvider;
+
+use League\Container\Container;
+use PHPUnit_Framework_TestCase;
+use TomPHP\ConfigServiceProvider\DIConfigServiceProvider;
+
+final class DIConfigServiceProviderTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    protected function setUp()
+    {
+        $this->container = new Container();
+    }
+
+    public function testItIsAConfigurableServiceProvider()
+    {
+        $this->assertInstanceOf(
+            'TomPHP\ConfigServiceProvider\ConfigurableServiceProvider',
+            new DIConfigServiceProvider([])
+        );
+    }
+
+    public function testItSetsUpABasicService()
+    {
+        $config = [
+            'example_class' => [
+                'class' => 'tests\mocks\ExampleClass'
+            ]
+        ];
+
+        $this->container->addServiceProvider(new DIConfigServiceProvider($config));
+
+        $this->assertInstanceOf(
+            'tests\mocks\ExampleClass',
+            $this->container->get('example_class')
+        );
+    }
+
+    public function testItCanCreateSingletonInstances()
+    {
+        $config = [
+            'example_class' => [
+                'class'     => 'tests\mocks\ExampleClass',
+                'singleton' => true,
+            ]
+        ];
+
+        $this->container->addServiceProvider(new DIConfigServiceProvider($config));
+
+        $instance1 = $this->container->get('example_class');
+        $instance2 = $this->container->get('example_class');
+
+        $this->assertSame($instance1, $instance2);
+    }
+
+    public function testItCanCreateUniqueInstances()
+    {
+        $config = [
+            'example_class' => [
+                'class'     => 'tests\mocks\ExampleClass',
+                'singleton' => false,
+            ]
+        ];
+
+        $this->container->addServiceProvider(new DIConfigServiceProvider($config));
+
+        $instance1 = $this->container->get('example_class');
+        $instance2 = $this->container->get('example_class');
+
+        $this->assertNotSame($instance1, $instance2);
+    }
+
+    public function testItDefaultsToCreatingUniqueInstances()
```

Comment:
@tomphp 

Maybe a bit ambiguous, as in fact, it does not provide the same instance, but a new one every time?